### PR TITLE
Use pagination when listing Digitalocean droplets

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
The DO provider used a nonsensical upper limit (1000), which is waaay above the actual DO hard limit. And the code would break if someone really had more than 1000 droplets in their account.

This PR implements very basic paging support for listing droplets.

```release-note
Fix Digitalocean if there are more than 200 droplets in the same account.
```

### CODE REVIEW

File: pkg/cloudprovider/provider/digitalocean/provider.go [hunk 1]
```diff
@@ -395,6 +392,35 @@ func (p *provider) get(machine *v1alpha1.Machine) (*doInstance, error) {
 	return nil, cloudprovidererrors.ErrInstanceNotFound
 }
 
+func (p *provider) listDroplets(token string) ([]godo.Droplet, error) {
+	ctx := context.TODO()
+	client := getClient(token)
+	result := make([]godo.Droplet, 0)
+	page := 1
```
Comment (thz) [inline pkg/cloudprovider/provider/digitalocean/provider.go]: https://github.com/digitalocean/godo
has an slightly simpler example for this paging. it does not start counting at 1 but 0.

Comment (xrstf) [reply pkg/cloudprovider/provider/digitalocean/provider.go]: Yeah I'll make it prettier, I was in a haste when I pushed this.

Comment (xrstf) [reply pkg/cloudprovider/provider/digitalocean/provider.go]: It's now a hypertext enabled HATEOAS client.

**Pull request merged**